cli/stork-publisher-agent/websocket.go

```diff
@@ -117,18 +118,34 @@ func (iwc *IncomingWebsocketConnection) Reader(priceUpdateCh chan PriceUpdate) {
 
 type OutgoingWebsocketConnection[T Signature] struct {
 	WebsocketConnection
+	assetIds                 map[AssetId]struct{}
+	assetIdsLock             sync.RWMutex
+	removed                  bool
 	logger                   zerolog.Logger
 	signedPriceUpdateBatchCh chan SignedPriceUpdateBatch[T]
 }
 
-func NewOutgoingWebsocketConnection[T Signature](conn WebsocketConnection, logger zerolog.Logger) *OutgoingWebsocketConnection[T] {
+func NewOutgoingWebsocketConnection[T Signature](conn WebsocketConnection, assetIds map[AssetId]struct{}, logger zerolog.Logger) *OutgoingWebsocketConnection[T] {
 	return &OutgoingWebsocketConnection[T]{
 		WebsocketConnection:      conn,
+		assetIds:                 assetIds,
 		signedPriceUpdateBatchCh: make(chan SignedPriceUpdateBatch[T], 4096),
 		logger:                   logger,
 	}
 }
 
+func (owc *OutgoingWebsocketConnection[T]) UpdateAssets(assetIds map[AssetId]struct{}) {
+	owc.assetIdsLock.Lock()
+	owc.assetIds = assetIds
+	owc.assetIdsLock.Unlock()
+}
+
+func (owc *OutgoingWebsocketConnection[T]) Remove() {
```

Comment:
The websocket may be `Closed` if there's an error, after which we'll reconnect. If the broker has been removed from the registry, we want to exit the read loop and not reconnect, so we'll call `Remove()`